python/ola/PidStoreTest.py

```diff
@@ -198,6 +198,79 @@ def testCmp(self):
     self.assertNotEqual(hash(p1b), hash(p2))
     self.assertNotEqual(hash(p1a), hash(p3))
 
+  def testPackUnpack(self):
+    store = PidStore.PidStore()
+    store.Load([os.path.join(path, "test_pids.proto")])
+
+    pid = store.GetName("DMX_PERSONALITY_DESCRIPTION")
+
+    # Pid.Pack only packs requests and Pid.Unpack responses
+    # so test in two halves
+    args = ["42"]
+    blob = pid.Pack(args, PidStore.RDM_GET)
+    decoded = pid._requests.get(PidStore.RDM_GET).Unpack(blob)[0]
+    self.assertEqual(decoded['personality'], 42)
+
+    args = ["42", "7", "UnpackTest"]
+    blob = pid._responses.get(PidStore.RDM_GET).Pack(args)[0]
+    decoded = pid.Unpack(blob, PidStore.RDM_GET)
+    self.assertEqual(decoded['personality'], 42)
+    self.assertEqual(decoded['slots_required'], 7)
+    self.assertEqual(decoded['name'], "UnpackTest")
+
+  def testPackRanges(self):
+    store = PidStore.PidStore()
+    store.Load([os.path.join(path, "test_pids.proto")])
+
+    pid = store.GetName("REAL_TIME_CLOCK")
+
+    args = ["2020", "6", "20", "20", "20", "20"]
+    blob = pid.Pack(args, PidStore.RDM_SET)
+    self.assertTrue(len(blob) > 1)
+
+    with self.assertRaises(PidStore.ArgsValidationError):
+      args = ["2000", "6", "20", "20", "20", "20"]
```

Comment:
I commented those cases for what they're testing in case someone needs to look at them again.  I sort of liked that one b/c the valid range isn't 0- or 1- so it seemed a bit more interesting to check boundaries....

Reply:
Yeah sounds good. I had to double-check the standard to confirm the minimum year was indeed 2003!